### PR TITLE
Add support for elasticsearch interface

### PR DESCRIPTION
**Pull request by PrasadG193**
Fixes: https://github.com/infracloudio/botkube/issues/53

ELS configuration structure:
```
  communications:
    # Settings for ELS
    elasticsearch:
      enable: false
      server: 'ELASTICSEARCH_ADDRESS'              # e.g https://example.com:9243
      username: 'ELASTICSEARCH_USERNAME'
      password: 'ELASTICSEARCH_PASSWORD'
      # ELS index settings
      index:
        name: botkube
        type: botkube-event
        shards: 1
        replicas: 0
```

Helm command to deploy botkube with ELS support:
```
helm install --name botkube --namespace botkube \
--set config.settings.clustername=dev-gke \
--set config.settings.allowkubectl=true \
--set config.communications.elasticsearch.enable=true \
--set config.communications.elasticsearch.server=ELS_ADDRESS \
--set config.communications.elasticsearch.username=ELS_USERNAME \
--set config.communications.elasticsearch.password=ELS_PASSWORD \
helm/botkube/
```

### CODE REVIEW

**Comment by PrasadG193** [comment]
image

File: pkg/filterengine/filters/image_tag_checker.go [hunk 1]
```diff
@@ -43,5 +43,5 @@ func (f *ImageTagChecker) Run(object interface{}, event *events.Event) {
 			event.Recommendations = append(event.Recommendations, ":latest tag used in image '"+c.Image+"' of Container '"+c.Name+"' should be avoided.\n")
 		}
 	}
-	log.Logger.Info("Image tag filter successful!")
+	log.Logger.Debug("Image tag filter successfull!")
```
Comment (ssudake21) [inline pkg/filterengine/filters/image_tag_checker.go]: successful

File: pkg/notify/elasticsearch.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+type index struct {
+	Shards   int `json:"number_of_shards"`
+	Replicas int `json:"number_of_replicas"`
+}
+
+// SendEvent sends event notification to slack
+func (e *ElasticSearch) SendEvent(event events.Event) error {
+	log.Logger.Debug(fmt.Sprintf(">> Sending to ElasticSearch: %+v", event))
+	ctx := context.Background()
+	client, err := elastic.NewClient(elastic.SetURL(e.Server), elastic.SetBasicAuth(e.Username, e.Password), elastic.SetSniff(false), elastic.SetHealthcheck(false), elastic.SetGzip(true))
```
Comment (ssudake21) [inline pkg/notify/elasticsearch.go]: are we doing auth for sending every event ?

Comment (ssudake21) [reply pkg/notify/elasticsearch.go]: What all options are available apart from basic auth ?

File: pkg/filterengine/filters/image_tag_checker.go (hunk 1 above)
Comment (PrasadG193) [reply pkg/filterengine/filters/image_tag_checker.go]: oops

File: pkg/notify/elasticsearch.go (hunk 2 above)
Comment (PrasadG193) [reply pkg/notify/elasticsearch.go]: We should initialize client during init() and reuse it. ElasticSearch doesn't have any other authentication method. The filebeat also uses basic auth for authentication
https://www.elastic.co/guide/en/beats/filebeat/current/elasticsearch-output.html

Comment (PrasadG193) [reply pkg/notify/elasticsearch.go]: Done!

**Comment by ssudake21** [comment]
@PrasadG193 Please make sure we also update documentation before this merge to master

**Comment by write2abhi** [comment]
How we can create indexes on a daily basis. Because It will generate a lot of data and we need to clear the data on daily basis?

**Comment by killcity** [comment]
Anyone else notice the **cluster** field not being populated in ES, but is correctly populated when calling Slack?

**Comment by aananthraj** [comment]
> cluster field not being populated in ES

@killcity thanks, 
#164 fixes this issue.

**Comment by killcity** [comment]
@aananthraj Thanks!

**Pull request merged**